src/Collection/CollectionTrait.php

```diff
@@ -330,7 +330,7 @@ public function sumOf($matcher = null)
      */
     public function shuffle(): CollectionInterface
     {
-        $elements = $this->toArray();
+        $elements = $this->toList();
```

Comment:
While this is for a bug fix, it's a change in behavior that could surprise people, so safer to target 4.1 and document in migration guide.

Reply:
I added it to 4.0.x in case we wanted to backport to 3.9.x. Up to you though.

Reply:
Dropping the keys is both good and bad. It will fix one but and potentially create another for existing users. One benefit of this change is it makes the behavior consistent with PHP's `shuffle()` method which also drops keys.

Reply:
All I can say is this is the current documentation for shuffle(). To me, that suggests that you want all the elements. If users need to have unique keys only, then they can re-create the collection first.

```
     * Returns a new collection with the elements placed in a random order,
     * this function does not preserve the original keys in the collection.
```